packages/next/build/webpack-config.ts

```diff
@@ -176,6 +177,113 @@ export default async function getBaseWebpackConfig(
 
   const devtool = dev ? 'cheap-module-source-map' : false
 
+  // Contains various versions of the Webpack SplitChunksPlugin used in different build types
+  const splitChunksConfigs: {
+    [propName: string]: webpack.Options.SplitChunksOptions
+  } = {
+    dev: {
+      cacheGroups: {
+        default: false,
+        vendors: false,
+      },
+    },
+    selective: {
+      cacheGroups: {
+        default: false,
+        vendors: false,
+        react: {
+          name: 'commons',
+          chunks: 'all',
+          test: /[\\/]node_modules[\\/](react|react-dom)[\\/]/,
+        },
+      },
+    },
+    prod: {
+      chunks: 'all',
+      cacheGroups: {
+        default: false,
+        vendors: false,
+        commons: {
+          name: 'commons',
+          chunks: 'all',
+          minChunks: totalPages > 2 ? totalPages * 0.5 : 2,
+        },
+        react: {
+          name: 'commons',
+          chunks: 'all',
+          test: /[\\/]node_modules[\\/](react|react-dom)[\\/]/,
+        },
+      },
+    },
+    prodGranular: {
+      chunks: 'all',
+      cacheGroups: {
+        default: false,
+        vendors: false,
+        framework: {
+          name: 'framework',
+          chunks: 'all',
+          test: /[\\/]node_modules[\\/](react|react-dom|scheduler|prop-types)[\\/]/,
+          priority: 40,
+        },
+        lib: {
+          test(module: { size: Function }): boolean {
+            return module.size() > 160000
+          },
+          name(module: { identifier: Function; rawRequest: string }): string {
+            const rawRequest = module.rawRequest
+            const identifier = module.identifier()
+            const trimmedIdentifier = /(?:^|\/)node_modules\/(.*)/.exec(
+              identifier
+            )
+            const processedIdentifier =
+              trimmedIdentifier &&
+              trimmedIdentifier[1].replace(/^@(\w+)\//, '$1-')
```

Comment:
This dasherizing still needs to happen for the `rawRequest` case, though. I can't speak to the Typescript, though.

Reply:
Whoops, yep. Added that in and also made the function return early if it finds a rawRequest